tests/lib/testing_classes.py

```diff
@@ -331,19 +335,24 @@ def publish_finished(self, dbcon, launched_app, download_test_data,
             print("Creating only setup for test, not launching app")
             yield False
             return
-        import time
+
         time_start = time.time()
         timeout = timeout or self.TIMEOUT
         timeout = float(timeout)
+
         while launched_app.poll() is None:
+            out, err = launched_app.communicate()
```

Comment:
Doesn't a call to `communicate` basically makes it idle/wait until the process finishes, making the time sleep, termination etc. to never be reached?

Should we instead do `launched_app.communicate(timeout=0.5)` and remove the sleep below it.
Note that you'd need to append all its outputs to retrieve the final result then.

Reply:
Thanks. Have changed this.